controllers/admin/AdminImportController.php

```diff
@@ -1260,161 +1805,180 @@ public function categoryImport($offset = false, $limit = false, &$crossStepsVari
             if ($validateOnly && isset($category->name) && ($category->parent == $category->name)) {
-                $this->errors[] = $this->trans(
-                    'A category can\'t be its own parent. You should rename it (current name: %1$s).',
+                $this->errors[] = $this->trans('A category can\'t be its own parent. You should rename it (current name: %1$s).',
                     array($category->parent),
                     'Admin.Advparameters.Notification'
                 );
                 return;
             }
-            $category_parent = Category::searchByName($id_lang, $category->parent, true);
-            if ($category_parent['id_category']) {
-                $category->id_parent = (int)$category_parent['id_category'];
-                $category->level_depth = (int)$category_parent['level_depth'] + 1;
+            $categoryParent = Category::searchByName($idLang, $category->parent, true);
+            if ($categoryParent['id_category']) {
+                $category->id_parent = (int)$categoryParent['id_category'];
+                $category->level_depth = (int)$categoryParent['level_depth'] + 1;
             } else {
-                $category_to_create = new Category();
-                $category_to_create->name = AdminImportController::createMultiLangField($category->parent);
-                $category_to_create->active = 1;
-                $category_link_rewrite = Tools::link_rewrite($category_to_create->name[$id_lang]);
-                $category_to_create->link_rewrite = AdminImportController::createMultiLangField($category_link_rewrite);
-                $category_to_create->id_parent = Configuration::get('PS_HOME_CATEGORY'); // Default parent is home for unknown category to create
-
-                if (($field_error = $category_to_create->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-                    ($lang_field_error = $category_to_create->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true &&
-                    !$validateOnly && // Do not move the position of this test. Only ->add() should not be triggered is !validateOnly. Previous tests should be always run.
-                    $category_to_create->add()) {
-                    $category->id_parent = $category_to_create->id;
+                $categoryToCreate = new Category();
+                $categoryToCreate->name = AdminImportController::createMultiLangField($category->parent);
+                $categoryToCreate->active = 1;
+                $categoryLinkRewrite = Tools::link_rewrite($categoryToCreate->name[$idLang]);
+                $categoryToCreate->link_rewrite = AdminImportController::createMultiLangField($categoryLinkRewrite);
+                // Default parent is home for unknown category to create :
+                $categoryToCreate->id_parent = Configuration::get('PS_HOME_CATEGORY');
+
+                // FIXME needing such a comment about "&& !$validateOnly" position is a code smell. Refacto needed.
```

Comment:
![i-heard-you-like-refacto-so-i-refacto-so-you-could-know-where-to-refacto](https://user-images.githubusercontent.com/6768917/29885221-fb268480-8dad-11e7-90a8-67f68abb2abf.jpg)

:laughing:

Reply:
Yeah, it was going beyond the 'tiny' refacto I was doing. But I wanted to leave a comment, not to forget later....